tests/server/mock/src/main/java/org/hawkular/apm/tests/server/ApmMockServer.java

```diff
@@ -239,13 +258,57 @@ public void handleRequest(final HttpServerExchange exchange) throws Exception {
                             exchange.getResponseSender().send(cc);
                         }
                     }
-                })).build();
+                }).addPrefixPath("v1/spans", new HttpHandler() {
+                    @Override
+                    public void handleRequest(HttpServerExchange exchange) throws Exception {
+                        if (exchange.isInIoThread()) {
+                            exchange.dispatch(this);
+                            return;
+                        }
+
+                        log.info("Zipkin spans request received: " + exchange);
+
+                        if (exchange.getRequestMethod() == Methods.POST) {
+                            exchange.startBlocking();
+
+                            String body = getBody(exchange.getInputStream());
+                            List<Span> spans = mapper.readValue(body, SPAN_LIST);
 
+                            synchronized (zipkinSpans) {
+                                zipkinSpans.addAll(spans);
+                            }
+
+                            exchange.getResponseHeaders().put(Headers.CONTENT_TYPE, "text/plain");
+                            exchange.getResponseSender().send("");
+                        } else if (exchange.getRequestMethod() == Methods.GET) {
+                            synchronized (zipkinSpans) {
+                                String zipkinSpansStr = mapper.writeValueAsString(zipkinSpans);
+                                exchange.getResponseHeaders().put(Headers.CONTENT_TYPE, "application/json");
+                                exchange.getResponseSender().send(zipkinSpansStr);
+                            }
+                        }
+                    }
+                })).build();
         server.start();
     }
 
     public void shutdown() {
         log.info("************ TEST TRACE SERVICE EXITING");
         server.stop();
     }
+
+    private String getBody(InputStream inputStream) throws IOException {
+        BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream));
```

Comment:
There's a neat trick for Java 8 and streams: `reader.lines().collect(Collectors.joining("\n"));`